### PR TITLE
SAK-43478 Lessons: Removed users can make the tool unusable

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java [hunk 1]
```diff
@@ -4954,7 +4954,9 @@ private void createEditTitleDialog(UIContainer tofill, SimplePage page, SimplePa
 				Collections.sort(siteMemberList, new Comparator<Member>() {
 					public int compare(Member lhs, Member rhs) {
 						UserSortNameComparator userComparator = new UserSortNameComparator();
-						return userComparator.compare(getUser(lhs.getUserId()), getUser(rhs.getUserId()));
+						Optional<User> lhsUserOptional = getUser(lhs.getUserId());
+						Optional<User> rhsUserOptional = getUser(rhs.getUserId());
+						return lhsUserOptional.isPresent() && rhsUserOptional.isPresent() ? userComparator.compare(lhsUserOptional.get(), rhsUserOptional.get()) : -1;
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java]: Shouldn't you return 0 if either of the optionals are not present? It seems more neutral to return a match of zero.

Comment (mpellicer) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java]: Ok

**Comment by mpellicer** [comment]
> It's likely better to adjust the comparator to be null safe after all null is a value.

Ok, I've implemented null checks in the comparator too.

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5248960&bid=17201994)

**Comment by ern** [comment]
@mpellicer see #8092 which is what I had in mind

**Pull request closed without merging**